mobile/src/main/java/org/openhab/habdroid/core/connection/CloudConnection.java

```diff
@@ -0,0 +1,19 @@
+package org.openhab.habdroid.core.connection;
+
+import android.support.annotation.NonNull;
+
+public class CloudConnection extends DefaultConnection {
+    static final String SETTINGS_ROUTE = "/api/v1/settings/notifications";
+
+    private String mSenderId;
+
+    public CloudConnection(@NonNull AbstractConnection baseConnection, @NonNull String senderId) {
+        super(baseConnection, TYPE_CLOUD);
+        mSenderId = senderId;
+    }
+
+    @NonNull
+    public String getMessagingSenderId() {
```

Comment:
I don't feel that this belongs into this class :/ It shouldn't be needed to cast a Connection object to a concrete implementation, just because so you get access to a specific method of this class. Can't we put this setting back somewhere else?

Reply:
This is a point where I *do* have strong feelings: IMHO it totally makes sense to put the sender ID into this class: it's a property of this specific server connection (as it was fetched from there) and it's only present if the  cloud connection is present (as it's fetched from the cloud API). Why would we want to introduce another random data holder here?
What's your proposed alternative solution?